marshal.go

```diff
@@ -983,6 +1000,23 @@ func (d *Decoder) unwrapPointer(mtype reflect.Type, tval interface{}, mval1 *ref
 	return mval, nil
 }
 
+func (d *Decoder) unmarshalText(tval interface{}, mval reflect.Value) error {
+	var s string
+	switch val := tval.(type) {
+	case string:
+		s = val
+	case bool:
+		s = fmt.Sprintf("%v", val)
+	case int64:
+		s = fmt.Sprintf("%d", val)
+	case float64:
+		s = fmt.Sprintf("%f", val)
+	default:
+		return fmt.Errorf("unspported type: %t", val)
```

Comment:
I think you meant `%T`?

Reply:
No longer needed with your other suggestion